rtl/include/eclhelper.hpp

```diff
@@ -2954,7 +2955,7 @@ struct IGlobalCodeContext
     virtual void fail(int, const char *) = 0;  
 
     virtual bool isResult(const char * name, unsigned sequence) = 0;
-    virtual unsigned getWorkflowId() = 0;
+    virtual unsigned getWorkflowId() const = 0;
```

Comment:
related to comment in thorcommon.hpp.
Can we rename it to avoid confusion? If not, is changing it's prototype from non-const to const problematic (does it affect VMT alignment of other members. I don't think so, but would like @ghalliday to comment on this and previous comment.

Reply:
IGlobalCodeContext  had getWorkflowId() defined before this PR.  I added getWorkflowId() to ICodeContext in this PR.  Some classes such as EclAgent derive from both IGlobalCodeContext  and ICodeContext.  Would it make sense to have both const and non-const defined in the classes such as EclAgent?

Reply:
yes, I realize it was already there, but now with a new method in ICodeContext, the pre-existing IGlobalCodeContext version (especially if never used) adds to the confusion.

>Would it make sense to have both const and non-const defined in the classes such as EclAgent?

I don't think it would be a good idea to have a const and non-const version, that would be more confusing.

I think IGlobalCodeContext ::getWorkflowid() could be deprecated.
[ see other comment : https://github.com/hpcc-systems/HPCC-Platform/pull/18232#discussion_r1457347909 ]

Reply:
I don't think changing from const to non const will cause any problems.  The danger is with failing to load workunit dlls that have references to old base definitions, but that should not happen with anything within IGlobalCodeContext.

IGlobalCodeContext ::getWorkflowid() was deprecated because it was incompatible with the parallel workflow execution.  This PR would have probably been a better way to get around that problem...